pkg/volume/iscsi/attacher.go

```diff
@@ -154,29 +155,62 @@ func (detacher *iscsiDetacher) UnmountDevice(deviceMountPath string) error {
 func (attacher *iscsiAttacher) volumeSpecToMounter(spec *volume.Spec, host volume.VolumeHost, pod *v1.Pod) (*iscsiDiskMounter, error) {
 	var secret map[string]string
 	var bkportal []string
-	iscsi, readOnly, err := getVolumeSource(spec)
+	readOnly, fsType, err := getISCSIVolumeInfo(spec)
 	if err != nil {
 		return nil, err
 	}
-	// Obtain secret for AttachDisk
-	if iscsi.SecretRef != nil && pod != nil {
-		if secret, err = volumeutil.GetSecretForPod(pod, iscsi.SecretRef.Name, host.GetKubeClient()); err != nil {
-			glog.Errorf("Couldn't get secret from %v/%v", pod.Namespace, iscsi.SecretRef)
+	if pod != nil {
+		secretNamespace, secretName, err := getISCSISecretInfo(spec, pod.Namespace)
+
+		if err != nil {
 			return nil, err
 		}
+
+		// Obtain secret for AttachDisk
+		if len(secretNamespace) > 0 && len(secretName) > 0 {
+			// if secret is provideded, retrieve it
+			kubeClient := host.GetKubeClient()
+			if kubeClient == nil {
+				return nil, fmt.Errorf("Cannot get kube client")
+			}
+			secretObj, err := kubeClient.Core().Secrets(secretNamespace).Get(secretName, metav1.GetOptions{})
+			if err != nil {
+				err = fmt.Errorf("Couldn't get secret %v/%v err: %v", secretNamespace, secretName, err)
+				return nil, err
+			}
+			secret = make(map[string]string)
+			for name, data := range secretObj.Data {
+				glog.V(12).Infof("retrieving CHAP secret name: %s", name)
```

Comment:
V(12) ?

Reply:
debug only

Reply:
no one could reasonably run at this level... V(5) or V(6) is the highest you'd probably want to go